core/generics/lo/lo.go

```diff
@@ -60,6 +61,18 @@ func KeyBy[K comparable, V any](collection []V, iteratee func(V) K) map[K]V {
 	return result
 }
 
+// KeyOnlyBy transforms a slice or an array of structs to a map containing the keys based on a pivot callback.
+func KeyOnlyBy[K comparable, V any](collection []V, iteratee func(V) K) map[K]types.Empty {
```

Comment:
This function resambles a lot the functional `map` (https://docs.python.org/3/library/functions.html#map) with the only difference that you are converting from a slice to a map.
Where are you using it?